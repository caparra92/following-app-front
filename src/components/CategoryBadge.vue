<template>
    <ion-row>
      <ion-col size="auto">
        <button>
          <ion-icon size = "large" :icon="addOutline"></ion-icon>
        </button>
      </ion-col>
      <ion-col>
        <p>{{ activityType }}</p>
      </ion-col>
      <ion-col size="auto">
        <button>
          <ion-icon size = "large" :icon="chevronForwardOutline"></ion-icon>
        </button>
      </ion-col>
    </ion-row>
</template>

<script setup>
import { IonGrid, IonRow, IonCol, IonIcon, IonButton } from '@ionic/vue';
import { addOutline, chevronForwardOutline } from 'ionicons/icons';
import { defineProps } from 'vue';


const props = defineProps({
  activityType: {
    type: String,
    default: "Activity type"
  }
});

</script>

<style scoped>

ion-row {
  margin: 1em 1.2em;
}

ion-col button {
  display: flex;
  justify-content: center;
  align-items: center;
  background: var(--ion-color-light);
  width: 44px;
  height: 44px;
  border-radius: 5px;
}

ion-col p {
  font-family: 'Outfit', sans-serif;
  text-transform: capitalize;
  font-size: 1.5em;
  margin: 0 auto;
}

ion-icon{
  color: var(--ion-color-dark);
}
</style>