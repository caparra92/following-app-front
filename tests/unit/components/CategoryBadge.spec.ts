import { mount, shallowMount } from '@vue/test-utils';
import CategoryBadge from '@/components/CategoryBadge.vue';

describe('Category Badge tests', () => {
    test('Verify HTML structure', () => {
        const wrapper = mount(CategoryBadge);
        expect(wrapper.element).toMatchSnapshot();
    });

    test('Verify if activityType has a default value', () => {
        const wrapper = mount(CategoryBadge);
        const p = wrapper.find('p');
        expect(p.text()).toBe('Activity type');
    });
})