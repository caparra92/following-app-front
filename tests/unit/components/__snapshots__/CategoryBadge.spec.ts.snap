// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Badge tests Verify HTML structure 1`] = `
<ion-row
  class=""
>
  <ion-col
    class=""
  >
    <button>
      <ion-icon />
    </button>
  </ion-col>
  <ion-col
    class=""
  >
    <p>
      Activity type
    </p>
  </ion-col>
  <ion-col
    class=""
  >
    <button>
      <ion-icon />
    </button>
  </ion-col>
</ion-row>
`;
